Зробив дз №2 перша перевірка

yarikgorpl opened the pull request:
Зробив,але мало зрозумів(((
Перевір,будьласка

yarikgorpl committed:
Зробив дз №2 перша перевірка

KhYulian commented on helpers/HttpError.js:
```diff
@@ -0,0 +1,7 @@
+  const error = new Error(message);
+  error.status = status;
+  return error;
+};
```
Використовуйте класи https://javascript.info/custom-errors, використовуйте new при роботі з класами

KhYulian commented on helpers/ctrlWrapper.js:
```diff
@@ -0,0 +1,12 @@
+      next(error);
+    }
+  };
+  return funcWrapper;
```
```suggestion
  return async (req, res, next) => {
    try {
      await ctrl(req, res, next);
    } catch (error) {
      next(error);
    }
  };
```

KhYulian commented on middlewares/validateBody.js:
```diff
@@ -0,0 +1,14 @@
+    }
+    next();
+  };
+  return func;
```
```suggestion
  return (req, res, next) => {
    const { error } = schema.validate(req.body);
    if (error) {
      next(HttpError(400, error.message));
    }
    next();
  };
```

KhYulian commented on models/contacts.js:
```diff
@@ -1,19 +1,57 @@
+  const contactIdToString = String(contactId);
+  const contacts = await listContacts();
+  const result = contacts.find((item) => item.id === contactIdToString);
+  console.log(result);
```
```suggestion
```

KhYulian commented on schemas/contacts.js:
Виносьте схеми валідації окремо e.g. /utils/validation/contactValidationSchemas.js

KhYulian commented on schemas/contacts.js:
```diff
@@ -0,0 +1,9 @@
+const Joi = require("joi");
+
+const addSchema = Joi.object({
```
```suggestion
const addContactSchema = Joi.object({
```

KhYulian reviewed (changes requested):
Залишив коменти

yarikgorpl committed:
Update schemas/contacts.js

Co-authored-by: Yulian <87574964+KhYulian@users.noreply.github.com>

yarikgorpl committed:
Update models/contacts.js

Co-authored-by: Yulian <87574964+KhYulian@users.noreply.github.com>

yarikgorpl committed:
Update middlewares/validateBody.js

Co-authored-by: Yulian <87574964+KhYulian@users.noreply.github.com>

yarikgorpl committed:
Update helpers/ctrlWrapper.js

Co-authored-by: Yulian <87574964+KhYulian@users.noreply.github.com>

yarikgorpl committed:
Виправив помилки

yarikgorpl committed:
Merge branch 'hw02-express' of https://github.com/yarikgorpl/nodejs-homework-rest-api into hw02-express

yarikgorpl committed:
Змінив експорт

yarikgorpl committed:
Fixed

Виправив помилку при якій під час оновлення в JSON записувався contactId замість id

KhYulian commented on helpers/HttpError.js:
```diff
@@ -0,0 +1,13 @@
+class ApiError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = ApiError;
+  }
+}
```
Додали, але вирішили не використовувати? : /
